distributor/billing.go

```diff
@@ -0,0 +1,33 @@
+package distributor
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"time"
+
+	billing "github.com/weaveworks/billing-client"
+	"github.com/weaveworks/common/user"
+)
+
+func (d Distributor) Emit(ctx context.Context, buf []byte, samples int64) error {
+	userID, err := user.Extract(ctx)
+	if err != nil {
+		return err
+	}
+
+	now := time.Now().UTC()
+	hasher := sha256.New()
+	hasher.Write(buf)
```

Comment:
Buf is the raw protos they sent up.  They could send up the same buf multiple times, but further up the stack that will get rejected due to timestamps going backwards.  It should be pretty unique per user, as it has a bunch of timestamps in it.